app/Models/BackupFile.php

```
@@ -87,6 +89,65 @@ public function isLocal(): bool
+
+        $targetVersion = static::normalizeVersion($service->installed_version ?: $service->version);
+        if ($targetVersion === null) {
+            return "Cannot determine the target server's database version.";
+        }
+
+        if (! static::versionGte($targetVersion, $this->database_version)) {
+            return "The target database version ({$targetVersion}) is lower than the backup source version ({$this->database_version}).";
+        }
+
+        return null;
+    }
```

PHPMD (2.15.0)

[warning] The method restoreCompatibilityError() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)